python/pyproject.toml

```
@@ -43,25 +43,25 @@ include = [{ path = "dev", format = "sdist" }]
 
 [tool.poetry.dependencies]
 python = "^3.8"
-mmhash3 = "3.0.1"
-requests = ">=2.28.1,<=2.28.2"
-click = "8.1.3"
-rich = ">=13.0.0,<=13.3.4"
-pyyaml = ">=5.4.0,<=6.0.0"
-pydantic = "1.10.7"
-fsspec = ">=2022.8.2,<=2023.3.0"
-pyparsing = "3.0.9"
-zstandard = "0.20.0"
-pyarrow = { version = ">=8.0.0,<=11.0.0", optional = true }
-pandas = { version = ">=1.4.4,<=1.5.3", optional = true }
-duckdb = { version = ">=0.6.0,<=0.7.1", optional = true }
-ray = { version = ">=2.0.0,<=2.3.1", optional = true }
-python-snappy = { version = "0.6.1", optional = true }
-thrift = { version = "0.16.0", optional = true }
-boto3 = { version = "1.24.59", optional = true }
-# The versions of the fsspec implementations should run in sync with fsspec above
-s3fs = { version = ">=2022.8.2,<=2023.3.0", optional = true }
-adlfs = { version = ">=2022.8.2,<=2023.3.0", optional = true }
+mmhash3 = "^3" # JUST: Unless 3.0.0's lack of Python 3.10/3.11 builds means it's a broken release, in which case ">=3.0.1, <4.0.0" may make sense, but it seems tests pass with 3.0.0.
+requests = "^2.20" # JUST: Tests pass with 2.22 (lower versions aren't compatible with some other strictly-pinned dependencies); picked nearby 2.20 as the most recent version without security risks.
```

Can you make sure that all the bound also have upper bounds?
```suggestion
requests = "^2.20,<3" # JUST: Tests pass with 2.22 (lower versions aren't compatible with some other strictly-pinned dependencies); picked nearby 2.20 as the most recent version without security risks.
```

---

They already do! Caret requirements work like this by default: https://python-poetry.org/docs/dependency-specification/#caret-requirements

---

Thanks, I learned something today. Some context: I was suggesting this because I like Dependabot to bump the versions: https://github.com/apache/iceberg/pull/7309/files#diff-1a45bd938160f7761769e7b6fbc819d1cc7f84ce37f1a6a23ef83cc4e214abc8L56 This way we can dependabot create PRs every week, and see if we're still compatible with the latest version.

---

I'm happy to change the syntax if you think it will help with Dependabot, but I expect (with the loosened dependencies) that you'll update the upper bound much less frequently.

---

@deepyaman That would be great! Having fewer notifications is good, as long as we're being notified of the major changes, so we can check if there are any API changes that break compatibility.

---

Done!